src/vinylcontrol/vinylcontrolprocessor.cpp

```diff
@@ -94,13 +94,8 @@ void VinylControlProcessor::run() {
             FIFO<CSAMPLE>* pSamplePipe = m_samplePipes[i];
 
             if (pSamplePipe->readAvailable() > 0) {
-                int samplesRead = pSamplePipe->read(m_pWorkBuffer, MAX_BUFFER_LEN);
-
-                if (samplesRead % 2 != 0) {
-                    qWarning() << "VinylControlProcessor received non-even number of samples via sample FIFO.";
-                    samplesRead--;
-                }
-                int framesRead = samplesRead / 2;
+                double samplesRead = pSamplePipe->read(m_pWorkBuffer, MAX_BUFFER_LEN);
+                double framesRead = samplesRead / 2;
```

Comment:
Why has this check been removed? Should be VERIFY_OR_DEBUG_ASSERT

Reply:
Perhaps I was told to do so. Earlier, frames were stored as integers, but now that we store frames as floating-point, the even check is not required.